sql/session.go

```diff
@@ -272,7 +286,7 @@ func NewContext(
 	ctx context.Context,
```

Comment:
Worried that this code can be called with a *sql.Context, since it is a context.Context. In that case, seems like it should inherit the view registry etc. Thoughts?

Reply:
I think the same issue exists with the Tracer, and the MemoryManager and the Session already.  I guess the biggest problem is that these methods (WithTracer, WithSesson, WithIndexRegistry, etc) return a closure with the ContextOption signature.  When evaluating each of these you have no idea which other ones have been provided.  Concretely:

    sqlCtx2 := sql.NewContext(sqlCtx,
        sql.WithIndexRegistry(NewIndexRegistry()),
        sql.WithTracer(sqlCtx),
        sql.WithViewRegistry(NewViewRegistry()),
    )

When it evaluates the WithTracer method it has no idea that Index and View Registries have been provided explicitly.

Reply:
OK, I think we can punt on this for now. Doesn't seem to actually be invoked anywhere outside of tests. I would put a warning comment on it for now.